lib/internal/http2/util.js

```diff
@@ -415,16 +417,18 @@ function mapToHeaders(map,
       if (isArray) {
         for (var k = 0; k < value.length; k++) {
           val = String(value[k]);
-          ret.push([key, val]);
+          ret += `${key}\0${val}\0`;
         }
+        count += value.length;
       } else {
         val = String(value);
-        ret.push([key, val]);
```

Comment:
Or, alternatively... (just brainstorming here ;-) ...)

```js
{
  ':status': 200,
  'foo': 'bar',
  [http2.noStore]: {
    'content-type': 'text/plain'
  }
}
```
Where `'http2.noStore` is a `Symbol`. This still feels weird, but it eliminates the potential polymorphism in the default case.

The other option is to change the headers API entirely and use a `Headers` object... e.g.

```js
const headers = new http2.Headers();
headers.set(':status', 200);
headers.set('content-type', 'text/plain', { store: false });
headers.set('foo', 'bar');
```

This has it's whole own set of issues, including being slower than using an object literal.